### ISSUE TITLE
Button A11y Pass

### ISSUE THREAD

**Issue by TestAccessibilityGIT** [OfficeDev/office-ui-fabric-react#6356]
Tracking all of the issues brought up with the 09/3 accessibility pass on Button control
Details on individual bugs can be found on [SharePoint](https://microsoft-my.sharepoint.com/:x:/r/personal/v-amaver_microsoft_com/_layouts/15/Doc.aspx?sourcedoc=%7B167e904b-6bd2-4bce-a1b7-1466fadf5c47%7D&action=default&uid=%7B167E904B-6BD2-4BCE-A1B7-1466FADF5C47%7D&ListItemId=1308&ListId=%7B395E1ECB-5002-440E-81C5-6A090E98077C%7D&odsp=1&env=prodbubble)

- [x] Bug 1 :[Accessibility][Screen reader]: Narrator does not read notification alert properly.
- [x] Bug 2 :[Accessibility][Screen reader]: In Scan mode user is not able to close the 'Sub menu' using 'Esc key'.
- [x] Bug 3 :[Accessibility][Keyboard Navigation]: Tab Focus is moving out of the Notification window which appears on activating Create account' control in 'Command Bar Button' .
- [x] Bug 4 :[Accessibility][Screen reader]: In back navigation on 'Email message menu item' narrator reads some other controls information in Scan mode.  
- [x] Bug 5 :[Accessibility][Usable]: Narrator focus moves twice on the first element of collapsed and split buttons in scan mode. 
- [x] Bug 6 :[Accessibility][Usable]: Screen does not scroll down along with keyboard focus in back navigation. 
- [x] Bug 7 :[Accessibility][Screen reader]: Tooltip information is not announced by the screen reader.
- [x] Bug 8 :[Accessibility][Screen reader]: Narrator scan mode focus moves separately onto the content inside the button under compound button section.
- [x] Bug 9 :[Accessibility][Screen reader]: User is not able to access the secondary action for the split buttons in scan mode.
- [x] Bug 10 :[Accessibility][Programmatic access]: Incorrect role is defined for 'Create account' split button control in 'Command Bar Button' using tab key. "
- [x] Bug 11: [Accessibility][Programmatic access]: Narrator does not convey about the disabled state of the primary action of 'split Button'.  "
- [x] Bug 12 :[Accessibility][Supporting the platform]: Some buttons are not visible in 'High contrast black and white theme setting'.
- [x] Bug 13 :[Accessibility][Usable]: Scan mode focus is not visible on 'Custom Split Button'. 
- [x] Bug 14 :[Accessibility][Usable]: 'Custom Split button' is visible as disable control while it is active. 
- [x] Bug 15 :[Accessibility][Programmatic access]: Name of the 'Custom split button' is not defined.
- [x] Bug 16 :[Accessibility][Visual requirements]: Tooltip is not provided for 'Split button with icon' button.
- [x] Bug 17 :[Accessibility][Programmatic access]: 'ChevronDown' Buttons must have discernible text.
- [x] Bug 18 :[Accessibility][Programmatic access]: ARIA attributes must conform to valid name of split buttons.
- [x] Bug19: [Accessibility][Visual requirements]: Visual focus indicator is not properly visible on standard create account button under split button variant.
- [x] Bug20: [Accessibility][Supporting the platform]: Arrow mark next to create account button under Primary and  Primary action disabled under spilt button variant is not visible in 'High contrast black and white theme setting'.

Component: Button

@micahgodbolt

**Comment by dzearing** [comment]
We are replacing the button, but I'd like to make sure we investigate these issues to understand the issue and requirement and that the new button and examples have these resolved.

**Comment by dollykumari5** [comment]
Adding a new issue.

Bug19: [Accessibility][Visual requirements]: Visual focus indicator is not properly visible on standard create account button under split button variant.

**Comment by dollykumari5** [comment]
Bug 6 and Bug 14  from the above list of bugs are fixed.
Verified on Test Environment:
OS: Version: 1903(OS Build 18362.100)
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.12.0/#/examples/button
Browser: Edge

**Comment by dollykumari5** [comment]
Adding a new issue 
Bug20: [Accessibility][Supporting the platform]: Arrow mark next to create account button under Primary and  Primary action disabled under spilt button variant is not visible in 'High contrast black and white theme setting'.

**Comment by kenotron** [comment]
Most of these are targeting an old version of the site, so the tests are not valid anymore. But I am going through all these bugs on the new site as much as I can. I'm testing all the non-scan mode first (parallels & remote desktop isn't really working well with capslock for some reason on my box)

**Comment by kenotron** [comment]
Bug 5 - cannot repro anymore

**Comment by kenotron** [comment]
Note for the future person to read this: SplitButton has a keyboard shortcut of alt-down to open the menu. So, that's actually how one should open the contextual menu rather than an enter or space which triggers the command (click)

**Comment by kenotron** [comment]
- [ ] TODO: Bug 16 is a real bug - will file a new bug for this, but only for example code.
need to add `title="New item"` to iconProps

**Comment by kenotron** [comment]
- [ ] TODO: Bug 17 is also real - custom split button example should have an `aria-label` for the chevron down

**Comment by kenotron** [comment]
- [ ] TODO: Bug 19 is real in that split focus should have a outline inside the button like the standard button, but not visible in the split button case

**Comment by kenotron** [comment]
- [ ] TODO: Bug 20 is real: high contrast mode does not show the icon in the split buttons:

![image](https://user-images.githubusercontent.com/34725/65095682-13523100-d977-11e9-8bea-13600a40607a.png)

**Comment by kenotron** [comment]
I've filed individual bugs for the ones that we're taking on for this button a11y pass.
